platform/conf/conf.go

```diff
@@ -1297,3 +1297,104 @@ func AddSSHKeys(userdata *UserData, keys *[]agent.Key) *UserData {
 	}
 	return userdata
 }
+
+// AddUserToGroups add the user to the given groups
+func (c *Conf) AddUserToGroups(user string, groups []string) {
```

Comment:
I think this function should return an error if an incompatible kind of configuration is used (e.g. cloudconfig). Just to avoid a situation where this function silently does nothing.

Reply:
I just gave you an easy way out. :P